modules/custom/dkan_datastore/src/Service/Resource.php

```diff
@@ -168,7 +168,19 @@ private function getResourceFromFileFetcher($node, $runFileFetcher) {
 
     $json = $fileFetcher->getResult()->getData();
     $fileData = json_decode($json);
-    return new R($node->id(), $fileData->destination);
+    return new R($node->id(), $fileData->destination, $this->getMimeType($node));
+  }
+
+  /**
+   * Private.
+   */
+  private function getMimeType($node) {
+    if (!$node) {
```

Comment:
@thierrydallacroce Instead of this check, should we use a type hint?

Reply:
Sure